pkgs/development/python-modules/wxPython/4.0.nix

```diff
@@ -22,14 +35,43 @@ buildPythonPackage rec {
     sha256 = "35cc8ae9dd5246e2c9861bb796026bbcb9fb083e4d49650f776622171ecdab37";
   };
 
-  nativeBuildInputs = [
-    pkgconfig
-  ];
+  doCheck = false;
 
-  buildInputs = [
-    gtk3 libjpeg libtiff SDL gst-plugins-base libnotify freeglut xorg.libSM
-    which
-  ];
+  nativeBuildInputs = [ pkgconfig which doxygen wxGTK libX11 ];
+
+  buildInputs = [ libjpeg libtiff SDL
+      gst-plugins-base libnotify freeglut xorg.libSM ncurses
+      requests libpng gstreamer libX11
+      pathlib2
+      (wxGTK.gtk)
+  ]
+    ++ lib.optional openglSupport pyopengl;
+
+  hardeningDisable = [ "format" ];
+
+  DOXYGEN = "${doxygen}/bin/doxygen";
+
+  preConfigure = lib.optionalString (!stdenv.isDarwin) ''
+    substituteInPlace wx/lib/wxcairo/wx_pycairo.py \
+      --replace 'cairoLib = None' 'cairoLib = ctypes.CDLL("${cairo}/lib/libcairo.so")'
+    substituteInPlace wx/lib/wxcairo/wx_pycairo.py \
+      --replace '_dlls = dict()' '_dlls = {k: ctypes.CDLL(v) for k, v in [
+        ("gdk",        "${wxGTK.gtk}/lib/libgtk-x11-2.0.so"),
+        ("pangocairo", "${pango.out}/lib/libpangocairo-1.0.so"),
+        ("appsvc",     None)
+      ]}'
+  '';
+
+  buildPhase = ''
+  python build.py -v --use_syswx dox etg --nodoc sip build_py
```

Comment:
Please build a wheel using `bdist_wheel`. The `installPhase` then won't need to be overridden.

Reply:
Thanks @FRidh! When I add `python build.py bdist_wheel` to the end of buildPhase, I get:
```
File "setup.py", line 15, in <module>
    from setuptools                     import setup, find_packages
ModuleNotFoundError: No module named 'setuptools'
```
Per the manual, I tried adding `setuptools_scm` to `nativeBuildInputs` and `setuptools` to `propogatedBuildInputs`, but the error is unchanged.

Reply:
The one time I had an error like that I think something was wrong with how PYTHONPATH was set up (determined by comparing the failing version to something else maybe?), but I don't think I spent the time to figure it out.

Reply:
They are using a custom build system based on waf so I don't think our generic builders will be able to handle this.

Reply:
I worked on my own version of this PR a while ago, and was able to get `bdist_wheel` working using this: https://github.com/lopsided98/nixpkgs/commit/47786de202c4637e3e60db617f04c1fd19e05030#diff-5dbc383e8d8890c6c54121d966199558R31